### PR TITLE
Adds responsive FlexContainer component

### PR DESCRIPTION
**Pull request by kyleshike**
For each breakpoint xs, sm, md, lg, xl, xxl, we provide optional flex props. meaning that for any breakpoint, the following are configurable:
```
alignItems?: 'stretch' | 'center' | 'flex-start' | 'flex-end' | 'baseline' | 'initial' | 'inherit';
flexDirection?: 'column' | 'column-reverse' | 'row' | 'row-reverse';
flexWrap?: 'wrap' | 'no-wrap' | 'wrap-reverse';
justifyContent?: 'flex-start' | 'flex-end' | 'space-between' | 'space-around' | 'space-evenly' | 'center' | 'initial' | 'inherit'
gap?: 1 | 2 | 3 | 4 | 5;
width?: 'xs' | 'sm' | 'md' | 'lg';
```
This props are of course also just available to the component directly to facilitate the most common use case in which breakpoints aren't necessary at all.

### CODE REVIEW

**Comment by kyleshike** [comment]
@jasonbasuil @gabescarbrough please review this today

**Comment by gabescarbrough** [comment]
@kyleshike will do! Going to block out some time to check this out today

**Comment by gabescarbrough** [comment]
Can you add the video you had in the last PR? It is 404ing there now

**Comment by kyleshike** [comment]
Appreciate it.  I'd love to get 3.1 released today if possible as there a few things folks are waiting on, and I'd also love to get this into that release because it's awesome

**Comment by kyleshike** [comment]
@gabescarbrough: https://www.loom.com/share/f960d1d3e15649eb8fcb3759895e9933?sid=ed463711-6cc0-454d-979e-52c46703c7de

File: src/FlexContainer/FlexContainer.module.css [hunk 1]
```diff
@@ -0,0 +1,767 @@
...
+  gap: var(--synth-spacing-3);
+}
+
+.gap_4 {
+  gap: var(--synth-spacing-4);
+}
+
+.gap_5 {
+  gap: var(--synth-spacing-5);
+}
```
Comment (jasonbasuil) [inline src/FlexContainer/FlexContainer.module.css:112]: Not sure if this was discussed already with designers, but wondering if we'll need spacing higher than 5?

File: src/FlexContainer/FlexContainer.module.css [hunk 2]
```diff
@@ -0,0 +1,767 @@
...
+  max-width: 512px;
+}
+
+.width_md {
+  max-width: 640px;
+}
+
+.width_lg {
+  max-width: 896px;
+}
```
Comment (jasonbasuil) [inline src/FlexContainer/FlexContainer.module.css:128]: is it possible to use any variables that we have for Card widths if we're using it here?

File: src/FlexContainer/FlexContainer.module.css [hunk 3]
```diff
@@ -0,0 +1,767 @@
...
+    flex-direction: row-reverse;
+  }
+
+  .sm_flexWrap_wrap {
+    flex-wrap: wrap;
+  }
+
+  .sm_flexWrap_no {
+    flex-wrap: no-wrap;
+  }
```
Comment (jasonbasuil) [inline src/FlexContainer/FlexContainer.module.css:310]: ❓ suggestion: `sm_flexWrap_no-wrap`

Comment (jasonbasuil) [reply src/FlexContainer/FlexContainer.module.css:310]: goes for the other ones as well

File: src/FlexContainer/FlexContainer.module.css (hunk 2 above)
Comment (jasonbasuil) [reply src/FlexContainer/FlexContainer.module.css:128]: also for the other other width values below

File: src/FlexContainer/FlexContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    'flex-start' |
+    'flex-end' |
+    'space-between' |
+    'space-around' |
+    'space-evenly' |
+    'center' |
+    'initial' |
+    'inherit';
+  gap?: 1 | 2 | 3 | 4 | 5;
+  width?: 'xs' | 'sm' | 'md' | 'lg';
```
Comment (jasonbasuil) [inline src/FlexContainer/FlexContainer.tsx:24]: Could possibly add some comments on here to describe the props and show up in the docs:

```
/**
    Sets the gaps (gutters) between rows and columns using Synthesis spacing values
  */
  gap?: 1 | 2 | 3 | 4 | 5;
  /**
    Sets the width of the FlexContainer using similar values to Card sizes
  */
  width?: 'xs' | 'sm' | 'md' | 'lg';
```

File: src/FlexContainer/FlexContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  md?: FlexProps;
+  lg?: FlexProps;
+  xl?: FlexProps;
+  xxl?: FlexProps;
+}
+
+export function FlexContainer({
+  children,
+  className,
+  ...props
```
Comment (jasonbasuil) [inline src/FlexContainer/FlexContainer.tsx:41]: ❓ our older `Flex` component has an `as` [prop](https://github.com/user-interviews/ui-design-system/blob/77c5d41dd0cb13b0cf4465701891189d83f8000a/src/Flex/Flex.tsx#L12). Would it be helpful to have that flexibility to define the element that is used as the `FlexContainer` as well (or a set of elements we want to allow)?

File: src/FlexContainer/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export * from './FlexContainer';
```
Comment (jasonbasuil) [inline src/FlexContainer/index.ts:1]: Need to export from top level `index.ts` as well

**Review by jasonbasuil — COMMENTED** [review]
Loving it! The gap property is slick and definitely an upgrade from the `Flex` component. A few comments and questions but looking good.

File: src/FlexContainer/FlexContainer.module.css (hunk 3 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.module.css:310]: oops!

File: src/FlexContainer/FlexContainer.module.css (hunk 2 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.module.css:128]: yea good call

File: src/FlexContainer/FlexContainer.module.css (hunk 1 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.module.css:112]: I chatted w Kiley about this and did some grepping of rails-server.  It seems like this should be sufficient for now:
https://userinterviews.slack.com/archives/G01P02LNJTF/p1715010523934299

File: src/FlexContainer/FlexContainer.tsx (hunk 5 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.tsx:41]: Looking at rails-server, that prop isn't used at all on the Flex component.  I think we should leave it out for now until we come upon a legit need for it

File: src/FlexContainer/index.ts (hunk 6 above)
Comment (kyleshike) [reply src/FlexContainer/index.ts:1]: https://github.com/user-interviews/ui-design-system/pull/1223/commits/ca1ea95bff4d1b58f9e9429d0a374d1e8ca8166c

File: src/FlexContainer/FlexContainer.module.css (hunk 3 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.module.css]: [ce479e4](https://github.com/user-interviews/ui-design-system/pull/1223/commits/ce479e494ca706151f0afd079f1ca91f3cfb40ea)

File: src/FlexContainer/FlexContainer.module.css (hunk 2 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.module.css]: ce479e4

File: src/FlexContainer/FlexContainer.tsx (hunk 4 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.tsx]: eabe050

File: src/FlexContainer/FlexContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+export type FlexContainerProps = FlexProps & {
+  className?: string;
+  children: ReactNode;
+  xs?: FlexProps;
+  sm?: FlexProps;
+  md?: FlexProps;
+  lg?: FlexProps;
+  xl?: FlexProps;
+  xxl?: FlexProps;
```
Comment (gabescarbrough) [inline src/FlexContainer/FlexContainer.tsx:41]: **praise**: 🙌

This is a neat pattern

File: src/FlexContainer/FlexContainer.module.css [hunk 8]
```diff
@@ -0,0 +1,767 @@
+.flexContainer {
+  display: flex;
+  max-width: 100%;
+}
+
```
Comment (gabescarbrough) [inline src/FlexContainer/FlexContainer.module.css:5]: **thought**: 💭

The classes below look a lot like Tailwind-style utility classes

An alternative approach would have been to add these utility classes as global styles and then any component could use the flex properties via `className`

I'm not advocating switching to that. You've wrapped this in a tidy component interface that is discoverable via TS types. Just pointing out that solving "generic styles" leads you to the same styles regardless of how you expose it to devs.

File: src/FlexContainer/FlexContainer.module.css [hunk 9]
```diff
@@ -0,0 +1,767 @@
...
+.alignItems_stretch {
+  align-items: stretch;
+}
```
Comment (gabescarbrough) [inline src/FlexContainer/FlexContainer.module.css:8]: **nitpick**: 💅 

```suggestion
.align-items_stretch {
  align-items: stretch;
}
```

Should the class names be kebab-case like the CSS properties? That would be my instinct.

**Review by gabescarbrough — APPROVED** [review]
🚀 nice job. This is going to be a powerful tool to not have to write custom layout styles per component.

File: src/FlexContainer/FlexContainer.tsx (hunk 7 above)
Comment (kyleshike) [reply src/FlexContainer/FlexContainer.tsx:41]: Right?  I'm really interested to see if this works out well.  If it does, then we should be able to do a similar thing to any variant that we want to support breakpoints

**Pull request merged**